File: Comparar.java
public class Comparar {

    class Fecha implements Comparable<Fecha>{
        int dia, mes, ano;

        public int compareTo(Fecha f){
            if(ano != f.ano) return ano - f.ano;
            if(mes != f.mes) return mes - f.mes;
            if(dia != f.dia) return dia - f.dia;
            return 0;
        }
    }
}

File: CodigosPython/Comparar.py
class Fecha:
    def __lt__(self,f):
        if self.ano != f.ano:
            return self.ano - f.ano
        if self.mes != f.mes:
            return self.mes - f.mes
        if self.dia != f.dia:
            return self.dia - f.dia
        return 0
        
        